src/main/java/de/zalando/aruha/nakadi/service/EventPublisher.java

```diff
@@ -50,38 +50,37 @@ public EventPublisher(final TopicRepository topicRepository,
     public EventPublishResult publish(final JSONArray events, final String eventTypeName) throws NoSuchEventTypeException,
             InternalNakadiException {
         final EventType eventType = eventTypeCache.getEventType(eventTypeName);
-        final List<BatchItem> batch = initBatch(events);
+        final List<BatchItem> batch = BatchFactory.from(events);
 
         try {
             validate(batch, eventType);
-            enrich(batch, eventType);
             partition(batch, eventType);
+            enrich(batch, eventType);
```

Comment:
@rcillo Ricardo I know you are deeper into that than me. This change of order, can it break anything for us? Because I'm not sure.